packages/react-graphql/src/hooks/README.md

```diff
@@ -0,0 +1,94 @@
+# Using Apollo Hooks
+
+## `ApolloProvider`
+
+Before using the individual hooks, you will need to wrap your application with `ApolloProvider` at root of your React component tree.
+
+You can replace `react-apollo`'s [`ApolloProvider`](https://www.apollographql.com/docs/react/api/react-apollo#ApolloProvider) if you are only going to use Hooks.
+
+```tsx
+import React from 'react';
+import {render} from 'react-dom';
+
+import ApolloClient from 'apollo-client';
+import {ApolloProvider} from 'react-graphql';
+
+const client = new ApolloClient();
+
+function App() {
+  return (
+    <ApolloProvider client={client}>
+      <MyRootComponent />
+    </ApolloProvider>
+  );
+}
+
+render(<App />, document.getElementById('root'));
+```
+
+Or alternative wrap the entire tree under the original `ApolloProvider` to support the both apollo components and apollo hooks.
+
+```tsx
+import React from 'react';
+import {render} from 'react-dom';
+
+import ApolloClient from 'apollo-client';
+import {ApolloProvider} from 'react-apollo';
+import {ApolloProvider as ApolloHooksProvider} from 'react-graphql';
+
+const client = new ApolloClient();
+
+function App() {
+  return (
+    <ApolloProvider client={client}>
+      <ApolloHooksProvider client={client}>
+        <MyRootComponent />
+      </ApolloHooksProvider>
+    </ApolloProvider>
+  );
+}
+
+render(<App />, document.getElementById('root'));
+```
+
+## `useMutation`
+
+This hook accepts two arguments: the mutation document, and optionally, a set of options to pass to the underlying mutation. It will return a function to that when invoked will trigger the mutation.
+
+```tsx
+import React from 'react';
+import {Form, TextField, Button, Banner} from '@shopify/polaris';
+import {useQuery} from '@shopify/react-graphql';
+
+import createCustomerMutation from './graphql/CreateCustomerMutation.graphql';
+
+function CustomerDetail() {
+  const [name, setName] = React.useState('');
+  const createCustomer = useMutation(createCustomerMutation);
+
+  async function handleFormSubmit({fields}: FormData<Fields>) {
```

Comment:
I don't think this example actually works (`Fields` is not defined here right?) could we either get rid of the generic or set it up so that it will actually compile?